app/views/affiliates/_form.html.erb

```diff
@@ -1,16 +1,53 @@
-<%= simple_form_for(@affiliate) do |f| %>
-  <%= f.error_notification %>
-
-  <div class="form-inputs">
-    <%= f.input :name %>
-    <%= f.input :address %>
-    <%= f.input :phone_number %>
-    <%= f.input :email %>
-    <%= f.input :website %>
-    <%= f.input :twitter %>
-  </div>
-
-  <div class="form-actions">
-    <%= f.button :submit %>
+<% if @affiliate.errors.any? %>
+  <div id="error_explanation">
+    <h2>
+      <%= pluralize(@affiliate.errors.count, "error") %> prohibited
+      this affiliate from being saved:
+    </h2>
+    <ul>
+    <% @affiliate.errors.full_messages.each do |msg| %>
+      <li><%= msg %></li>
+    <% end %>
+    </ul>
   </div>
 <% end %>
+
+<form role="form">
+
+  <section id="contactInfo">
+    <h4>Information</h4>
+    <hr>
+    <div class="row">
+      <div class ="col-md-4">
+        <%= f.input :name, label: "Affiliate Name", maxlength: 255, required: true %>
+      </div>
+
+      <div class = "col-md-4">
+        <%= f.input :address, maxlength: 255, required: true %>
```

Comment:
Yeah I think it would be needed. An affiliate's city is not necessarily denoted by its name, so we would need a full address for them.